std/stdio.d

```diff
@@ -3084,11 +3084,24 @@ void main()
         import std.exception : collectException;
 
         ulong pos = void;
-        if (collectException(pos = tell)) return ulong.max;
+        if (collectException(pos = tell))
+            return ulong.max;
+
+        // Unfortunately there doesn't seem to be a reliable way to
+        // find out if a file is seekable without actually attempting
+        // to seek in it. Even seek(0, SEEK_CUR) will succeed in
+        // unseekable files.
+        if (collectException(seek(0, SEEK_END)))
```

Comment:
If you collect the exception here and don't throw it, doesn't that mean this will never throw and the docs are wrong?

Reply:
"Even seek(0, SEEK_CUR) will succeed in unseekable files."

This is because FILE * is buffered! I wonder if we attempted seeking the actual handle, if we wouldn't be better off. It would also allow us to avoid flushing the buffer for this check.

Reply:
you should put a flag if the file is not seek-able. In an OOP api non seek-able files such as pipes would be subclassed and throw on seek attempts.